pkg/sysregistriesv2/system_registries_v2.go

```diff
@@ -358,6 +358,16 @@ func ConfigPath(ctx *types.SystemContext) string {
 // ConfigDirPath returns the path to the system-wide directory for drop-in
 // registry configuration files.
 func ConfigDirPath(ctx *types.SystemContext) string {
+	if ctx != nil && ctx.RootForImplicitAbsolutePaths != "" {
+		return filepath.Join(ctx.RootForImplicitAbsolutePaths, systemRegistriesConfDirPath)
+	}
+
+	return systemRegistriesConfDirPath
+}
+
+// UserConfigDirPath returns the user's directory for drop-in
+// registry configuration files, returns "" is not specified
+func UserConfigDirPath(ctx *types.SystemContext) string {
```

Comment:
We’ve made the mistake of providing `ConfigDirPath` exposing internal logic, which is now unusable (we have up to two directories, but can only return one), once; let’s not do it again.

Please:
- Gather all the logic for deciding on paths to use into `newConfigWrapper`
- Make `ConfigPath` and `ConfigDirPath` call `newConfigWrapper` and extract the paths form from there
- Provide new `ConfigurationSourceDescription`, which is documented to return a user-readable string to be possibly included in error messages (NOT paths otherwise consumable by callers), and returns all the paths from a `newConfigWrapper`.
- Document `ConfigPath` and `ConfigDirPath` as deprecated, directing callers to the new function.

That would be ideal as a refactoring / API change commit that does not change the actual logic (does not add the second directory), and then we can use a separate commit for adding the new features.